### DISCUSSION TITLE
Operational Attribute in ObjectFactory

### DISCUSSION THREAD

**Discussion by kellyethridge** [MarimerLLC/csla#1702] [Questions]
Hello,

I'm getting an error from an analyzer for putting an operational attribute on a `DataPortal` method in an class deriving from `ObjectFactory`. Are operational attributes supported within an object factory class?

Thank you.

**Comment by rockfordlhotka** [comment]
Not at this time. Perhaps not ever, though I'm not sure and would appreciate your input.

Right now the method names are provided via the attribute, and so _also_ providing them with attributes is redundant and could be confusing.

I suppose I could mark the method name options in the attribute as `Obsolete` and force a change to the new model?

**Comment by kellyethridge** [comment]
I'm fine with not using the attributes in an `ObjectFactory`. I was just unaware that the attributes were not supported in that case.

**Reply by rockfordlhotka** [reply]
If you have comments or thoughts, [here's the work issue for modernizing ObjectFactory](https://github.com/MarimerLLC/csla/issues/1707).
